edge-hub/src/Microsoft.Azure.Devices.Routing.Core/endpoints/StoringAsyncEndpointExecutor.cs

```diff
@@ -175,65 +175,54 @@ void Dispose(bool disposing)
         // A pump is started as soon as the object is created, and it keeps the messages list populated.
         internal class StoreMessagesProvider
         {
+            static readonly IList<IMessage> EmptyList = new List<IMessage>();
             readonly IMessageIterator iterator;
             readonly int batchSize;
             readonly AsyncLock messagesLock = new AsyncLock();
-            readonly AsyncManualResetEvent messagesResetEvent = new AsyncManualResetEvent(true);
-            readonly TimeSpan timeout;
-            readonly Task populateTask;
-            List<IMessage> messagesList;
+            Option<Task<IList<IMessage>>> getMessagesTask;
```

Comment:
I don't see when this would ever be None?

Reply:
Good catch, initially I was prefetching the messages only when there were more to be fetched, but later changed the logic. Can remove the Option and the empty list...